blocks/flip-cards/flip-cards.js

```diff
@@ -1,16 +1,67 @@
-export default function decorate(block) {
+async function getHtml(url) {
+  let html;
+  const page = await fetch(url);
+  if (page.ok) {
+    html = await page.text();
+  } else {
+    html = '';
+  }
+  const retEl = document.createElement('div');
+  retEl.innerHTML = html;
+  return retEl;
+}
+
+function getDocumentMetadata(name, document) {
+  const attr = name && name.includes(':') ? 'property' : 'name';
+  const meta = [...document.querySelectorAll(`meta[${attr}="${name}"]`)]
+    .map((m) => m.content)
+    .join(', ');
+  return meta || '';
+}
+
+export default async function decorate(block) {
   const cols = [...block.firstElementChild.children];
+  const rows = [...block.children];
+  const fronts = [];
+  const backs = [];
+  const links = [];
+
   block.classList.add(`.flip-cards-${cols.length}-cols`);
 
   if (block.children.length === 0) {
     block.innerHTML = 'Malformed block structure. No content.';
     return;
   }
 
-  const rows = [...block.children];
-  const fronts = [...rows[0].children];
-  const backs = rows[1] ? [...rows[1].children] : [];
-  const links = rows[2] ? [...rows[2].children] : [];
+  const pages = block.querySelectorAll('a');
+  const pageWorkQueue = [];
+  pages.forEach((page) => pageWorkQueue.push(getHtml(page.href)));
+
+  const doms = await Promise.all(pageWorkQueue);
+
+  doms.forEach((dom, idx) => {
+    const cardFront = document.createElement('div');
+    const cardBack = document.createElement('div');
+    const icon = rows[0].children.item(0);
+    const titleh3 = document.createElement('h3');
+    const titleh4 = document.createElement('h4');
+    titleh3.innerText = getDocumentMetadata('og:title', dom);
+    titleh4.innerText = titleh3.innerText;
+    const desc = document.createElement('p');
+    desc.innerText = getDocumentMetadata('og:description', dom);
+    const link = pages[idx];
+
+    if (icon && titleh3 && titleh4 && desc && link) {
```

Comment:
I think we need to rethink this condition as well, since the cards could technically also work if not everything is present. IMO it would be fine if no icon is present for example and we don't need to check for both titles. Maybe also flip-cards without a link would be feasible, however that doesn't really make sense for the automatically fetched ones...

Reply:
Done! I check for all of the individual items; add them; otherwise we still append the items to the front, back, etc resulting in empty cards if that's how authors like it.